tests/integration/replication-4.tcl

```diff
@@ -195,3 +195,45 @@ start_server {tags {"repl external:skip"}} {
         }
     }
 }
+
+start_server {tags {"repl external:skip"}} {
+    start_server {} {
+        set master [srv -1 client]
+        set master_host [srv -1 host]
+        set master_port [srv -1 port]
+        set replica [srv 0 client]
+
+        test {First server should have role slave after SLAVEOF} {
+            $replica slaveof $master_host $master_port
+            wait_for_condition 50 100 {
+                [s 0 role] eq {slave}
+            } else {
+                fail "Replication not started."
+            }
+            wait_for_sync $replica
+        }
+
+        test {Data divergence can happen under default conditions} {       
+            $replica config set repl-error-behavior ignore     
+            $master debug replicate fake-command-1
+
+            # Wait for replication to normalize
+            $master set foo bar2
+            $master wait 1 2000
+
+            assert_equal [count_log_message 0 "fake-command-1"] 1
```

Comment:
maybe add a comment that you're looking for the `== CRITICAL ==` print.

p.s. did you grep the logs with `--dont-clean` to make sure it doesn't happen anywhere?
even with the default explicitly set to `panic`, there's a chance this issue happens and for some reason the tests don't fail despite the crash .

Reply:
I did also grep for the logs and didn't find the '== CRITICAL =='. Will also add the check for critical here.